modules/calc/src/main/java/com/opengamma/strata/calc/runner/TaskBuilder.java

```diff
@@ -0,0 +1,204 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.calc.runner;
+
+import java.util.List;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ListMultimap;
+import com.opengamma.strata.basics.CalculationTarget;
+import com.opengamma.strata.calc.CalculationRunner;
+import com.opengamma.strata.calc.marketdata.mapping.MarketDataMappings;
+import com.opengamma.strata.calc.runner.function.CalculationFunction;
+import com.opengamma.strata.collect.ArgChecker;
+
+/**
+ * Builder used to create calculation tasks.
+ * <p>
+ * The {@link CalculationRunner} produces a grid of results, with a row for each target
+ * and a column for each measure. The targets and columns that define the grid of results
+ * are passed in using an instance of {@code CalculationTasks}.
+ * This builder is used to produce the tasks for a single target.
+ * <p>
+ * The result of the builder consists of a number of tasks, each containing one or more task cells.
+ * Each task cell corresponds to a single column in the row.
+ */
+public final class TaskBuilder {
```

Comment:
This is an odd class. It doesn't follow the normal pattern for a builder. It only has one mutation method which is `executeSeparately`, otherwise all the data is passed into the constructor. And it has getters which is a departure for a builder.

Separate execution seems like it would be a natural property of the column as it's the measure which dictates how long it takes to calculate. If `executeSeparately` were remove then this whole class could be an immutable bean.

Are the getters necessary? Could it be reduced to a static method returning a list of tasks?